pkg/istio-agent/xds-proxy.go

```diff
@@ -155,6 +157,7 @@ func (p *XdsProxy) StreamAggregatedResources(downstream discovery.AggregatedDisc
 			}
 		}
 	}()
+	go p.healthChecker.PerformApplicationHealthCheck(healthCheckRequestChan)
```

Comment:
I would prefer the xds-proxy is has minimal application logic such as the health check. Instead it can expose some method like SendDiscoverRequest that the health checker calls, and otherwise doesn't need to be coupled. We can probably replace `ndsRequestChan` with a generic name like `directRequestChan` or something and reuse that one.

Reply:
where else would the health check logic occur? The goroutine needs to start somewhere, right?

I think we can expose a `SendDiscoverRequest` method, but im not sure where the healthchecker would start if not xds-proxy. would it be istio-agent?